jupyterhub_moss/spawner.py

```diff
@@ -351,70 +342,103 @@ def options_from_form(self, formdata: Dict[str, List[str]]) -> Dict[str, str]:
 
         self.__validate_options(options)
 
-        partition = options["partition"]
+        partition_environments = tuple(
+            self.partitions[options["partition"]]["jupyter_environments"].values()
+        )
+        if "environment_path" not in options:
+            # Set path to use from first environment for the current partition
+            options["environment_path"] = partition_environments[0]["path"]
+
+        corresponding_default_env = find(
+            lambda env: env["path"] == options["environment_path"],
+            partition_environments,
+        )
+        # Singularity images are never added to PATH,
+        # custom envs are always added to PATH, defaults ones only if add_to_path is True
+        if not options["environment_path"].endswith(".sif") and (
+            corresponding_default_env is None
+            or corresponding_default_env["add_to_path"]
+        ):
+            options["prologue"] = f"export PATH={options['environment_path']}:$PATH"
+
+        return options
+
+    def __check_user_options(self, partition_info):
```

Comment:
"Dynamic" checks only.
Exceptions raised here leads to "500 Internal Server Error" without explanations

Reply:
Again, worth adding to the docstring/comment